src/Lecture3.hs

```diff
@@ -125,10 +162,16 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) :: Reward -> Reward -> Reward
+  Reward {rewardGold=g1, rewardSpecial=s1} <> Reward {rewardGold=g2, rewardSpecial=s2} = 
+    Reward {rewardGold=(g1 <> g2), rewardSpecial=(s1 || s2)}
 
 instance Monoid Reward where
-
+  mempty :: Reward
+  mempty = Reward 
+    { rewardGold = Gold 0
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
    { rewardGold = mempty
```